src/Jmikola/AutoLogin/Http/Firewall/AutoLoginListener.php

```diff
@@ -30,20 +28,19 @@ class AutoLoginListener implements ListenerInterface
     /**
      * Constructor.
      *
-     * @param TokenStorageInterface|SecurityContextInterface $securityContext
+     * @param TokenStorageInterface                          $securityContext
```

Comment:
If we tag 2.0 after merging this PR, would you agree that we can simply add TokenStorageInterface to the constructor type hint and remove the explicit `instanceof` check for throwing an InvalidArgumentException?

Looking at [`security.xml`](https://github.com/jmikola/JmikolaAutoLoginBundle/blob/75761f28bc4dc69ead9380b1334d72365cfcd06b/Resources/config/security.xml#L15) in the bundle, the argument is not defined but later [replaced in the factory](https://github.com/jmikola/JmikolaAutoLoginBundle/blob/75761f28bc4dc69ead9380b1334d72365cfcd06b/DependencyInjection/Security/AutoLoginFactory.php#L30) depending on the Symfony version being used. If the bundle also raises its Symfony requirement beyond 2.6, there should never be a case where a SecurityContextInterface is injected. It looks like you've also addressed that in your corresponding bundle PR: https://github.com/jmikola/JmikolaAutoLoginBundle/pull/26/files#diff-e429bab8ab61fdc51ae458fac147a4dbL30

FWIW, I think we need to tag 2.0 after merging this anyway since AlreadyAuthenticatedEvent no longer inheriting the same class is technically BC break (albeit unlikely to affect anyone's application).

Reply:
You are absolutely right, the typehint would be sufficient since this new version would introduce BC hence the new major version tag.